### PR TITLE
[Feature]support export on cn when run on shared_data mode

### PR DESCRIPTION
**Pull request by abc982627271**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 8 / 38 (21.05%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|0|6|00.00%|[491, 492, 587, 588, 589, 591]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|14|00.00%|[728, 729, 730, 731, 733, 734, 737, 738, 739, 740, 767, 768, 769, 771]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|10|00.00%|[112, 130, 131, 132, 135, 139, 140, 141, 145, 148]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java [hunk 1]
```diff
@@ -723,13 +724,21 @@ public Status releaseSnapshotPaths() {
...
+
+            ComputeNode node = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(host, port);
+            if (node == null) {
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                    node = GlobalStateMgr.getCurrentSystemInfo().getComputeNodeWithBePort(host, port);
+                }
+                if (node == null) {
+                    continue;
+                }
             }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java:736]:
```suggestion
            ComputeNode node = GlobalStateMgr.getCurrentSystemInfo(). getBackendOrComputeNodeWithBePort(host, port);
            if (node == null) {
                continue;
            }
```

File: fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java [hunk 2]
```diff
@@ -723,13 +724,21 @@ public Status releaseSnapshotPaths() {
...
-            long backendId = backend.getId();
-            if (!GlobalStateMgr.getCurrentSystemInfo().checkBackendAvailable(backendId)) {
-                continue;
+            long nodeId = node.getId();
+            if (!GlobalStateMgr.getCurrentSystemInfo().checkBackendAvailable(nodeId)) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java:738]: check node alive

File: fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java [hunk 3]
```diff
@@ -755,8 +764,11 @@ public Status releaseMetadataLocks() {
                 TNetworkAddress address = location.getServer();
                 String host = address.getHostname();
                 int port = address.getPort();
-                Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(host, port);
-                if (backend == null) {
+                ComputeNode node = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(host, port);
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                    node = GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNodeWithBePort(host, port);
+                }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java:770]:
```suggestion
                ComputeNode  node = GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNodeWithBePort(host, port);
```

File: fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java [hunk 4]
```diff
@@ -107,6 +108,8 @@ protected void exec() {
     }
 
     private Status makeSnapshots() {
+        // for debug
+        LOG.info("enter makeSnapshots()");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java:112]: remove

File: fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java [hunk 5]
```diff
@@ -124,18 +127,25 @@ private Status makeSnapshots() {
                 TNetworkAddress address = location.getServer();
                 String host = address.getHostname();
                 int port = address.getPort();
-                Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(host, port);
-                if (backend == null) {
+                ComputeNode node = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(host, port);
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                    node = GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNodeWithBePort(host, port);
+                }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java:133]:
```suggestion
                ComputeNode node = GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNodeWithBePort(host, port);
```

File: fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java [hunk 6]
```diff
@@ -124,18 +127,25 @@ private Status makeSnapshots() {
...
+
+                if (node == null) {
                     return Status.CANCELLED;
                 }
-                long backendId = backend.getId();
-                if (!GlobalStateMgr.getCurrentSystemInfo().checkBackendAvailable(backendId)) {
+
+                long nodeId = node.getId();
+                if ((RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING) &&
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java:140]: check node alive

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 8 / 25 (32.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|0|7|00.00%|[491, 492, 494, 589, 590, 591, 593]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|5|00.00%|[727, 728, 731, 758, 759]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[127, 128, 132, 133, 136]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 15 / 25 (60.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|5|00.00%|[727, 728, 731, 758, 759]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[127, 128, 132, 133, 136]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 15 / 22 (68.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[760, 761]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[127, 128, 132, 133, 136]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 15 / 22 (68.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[760, 761]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[127, 128, 132, 133, 136]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 17 / 22 (77.27%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[760, 761]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|2|5|40.00%|[132, 133, 136]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 20 / 22 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[760, 761]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|5|5|100.00%|[]|

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wyb** [comment]
@Mergifyio backport branch-3.2

**Comment by abc982627271** [comment]
@mergifyio backport branch-3.1

**Pull request merged**